packages/theming/theme-types/src/Theme.types.ts

```diff
@@ -19,7 +20,10 @@ export interface Theme {
   components: { [key: string]: object };
   componentTokens?: object;
   spacing: Spacing;
-  host: object; // platform specific host settings
+  host: {
+    // Office palette, if running in Office with the native module connected in the theme
+    palette?: OfficePalette;
```

Comment:
You don't have any quibbles about leaving this in type in a semi-official capacity?

Reply:
I've been debating this but I don't think so, though perhaps we rename it officePalette. I kind of envision host maybe eventually looking something like:

```ts
// host is basically a grouping for settings that come from native modules
host: {
  dark?: boolean;
  highContrast?: boolean;
  officePalette?: OfficePalette;
  win32?: Win32Settings;
  windows?: WindowsSettings;
  macos?: MacSettings;
  // or something like this
}
```

Then the types in ThemeReference are general so that someone can do the following:

```ts
type MyTheme = Theme & {
  host: {
    myAdditionalNativeSettings?: SomeType;
  }
}
```

Then basically fill those settings the same way the win32 theme does. You can imagine comments creating a theme, based off the Win32 theme, where they augment it with their native module but use our context and provider mechanism. Then we get rid of the dual context problem we've had in the past.

Reply:
+1 to allowing for flexible and general usage, but one thing I'd be interested in is keeping all these names generic (`officePalette` seems a little constraining maybe) - there could be a world in the future where the native modules are written in a way where they're pluggable. Other brownfield scenarios where large applications serve as "hosts", aren't too far-fetched a notion imo.

Reply:
Yeah, I tend to agree though the question is where that coalesces. There are several levels of definitions and naming.

- **Base** - need a good name but these are things like `theme.colors.neutralPrimary` or `theme.colors.neutral3` or whatever we come up with
- **Semantic** - these are role based.  Right now they are things like `theme.colors.bodyBackground` but may change. Design is working on a new set of these now, and while they feel like the base layer may be volatile, these names might be stable(ish)
- **Native/Host** - at some point there is a different concept of naming that maps to what exists in native. That can either be exposed in the native module, or on the theme somewhere, or be hidden on the native side.

I think there are a couple of principles I've been thinking about:

- components should be authored against the semantic level. It is kind of the goldilocks level here and allows for a role based mapping of arbitrary color schemes.
- this means that customizations get authored against the semantic level, which makes sense
- this also implies native needs to map to semantic, rather than mapping to the base layer, which is great because this maps to the way these things tend to be built on the native side.
- this native -> semantic mapping can be lossy, meaning that some native values may have no mapping in the semantic layer. 
- This implies that there will likely be value in having those available somewhere in the theme, because then a component can be customized easily to depend on those colors on a given platform.

This is all subject to debate and discussion but this is just the way I've been starting to think about things.

Reply:
I generally agree with the laid out principles. I would add, as a direct consequence of native -> semantic mapping be lossy + the probable usage of base when semantic is insufficient, that affordances for native -> base mappings are probably unavoidable. And I think the code already reflects this.

Reply:
Yea I think I'm _mostly_ on board with this strategy, although I think the one place I might object to the mapping of native -> semantic is with the brand/accent colors.

Reply:
Sent too soon --- I think you had concerns about this mapping being difficult to reason about because (currently) there are so many different accent colors in the ramp and we only use a subset--- I.e. if we prune out Accent 20-45, is it weird they we assign App2 to Accent 15 and App3 to Accent50?

I'm concerned that not mapping App -> Base layer here undercuts the flexibility of the Fluent theme and limits the ability to inject different brands.  Maybe this is something design should have a say in though.   Any thoughts about that?

Reply:
I definitely want design to iterate a bit more. From talking to Travis it sounds like their notion of base colors are more fluid than their notion of semantic ones. (They call them Sets) In general design might retarget the set value from Neutral.89 to Neutral.73 and we shouldn't rely on meaning for Neutral.73.

Because the set/semantic colors have explicit branded values that would be where we would map brand colors.